src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,105 @@
+import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../types';
+
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+                                                  commentsList, setCommentsList,
+                                                }) => {
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
+
+  const addComment = (newComment: Comment) => {
+    if (commentsList) {
+      commentsList.unshift(newComment);
+      const copy = [...commentsList];
+
+      setCommentsList(copy);
+    }
+  };
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (yourname && youremail && yourcomment && commentsList) {
+      const newComment = {
+        id: commentsList.length + 1,
+        postId: commentsList[0].postId,
+        name: yourname,
+        email: youremail,
+        body: yourcomment,
+      };
+
+      addComment(newComment);
+      setError(false);
+      setYourname('');
+      setYouremail('');
+      setYourcomment('');
```

Comment:
Your comment form is not actually adding the comment to the post, it's just being shown to the user in the screen. Instead you should call you comments api function which sends the request to post the comment in the api.

I believe you used this as a test to see if you would be able to see the comments, even though your post wasn't opening as intended.